### PR TITLE
Add GPT-OSS 20B CUDA support

### PR DESCRIPTION
**Pull request by gracee3**
## Summary

Add CUDA runtime support for `GptOssForCausalLM` / `openai/gpt-oss-20b`.

This PR:
- adds GPT-OSS config parsing and transport through engine -> worker -> runner
- adds a real GPT-OSS architecture module and GPT-OSS-specific weight plumbing
- preserves GPT-OSS tensor shapes and `U8` MXFP4 expert payloads during loading
- adds sink-aware alternating sliding/full attention support in the CUDA path
- adds executable GPT-OSS MXFP4 expert routing / expert projection support
- fixes the GPT-OSS flash-attention launch packing bug that caused illegal KV-cache reads at runtime
- fixes post-merge GPT-OSS runtime fallout in fp16 setup and PTX discovery so the merged CUDA path still boots and serves

## Why

`openai/gpt-oss-20b` was recognized but not executable. The missing pieces were:
- config/model metadata transport for GPT-OSS-specific fields
- weight loading that preserves MXFP4 expert tensors and shape metadata
- runtime support for sink logits and alternating sliding/full attention
- executable GPT-OSS MoE routing and MXFP4 expert projections
- correct runtime handling for GPT-OSS-specific decode constraints

The late runtime bug fixed in this stack was a host/kernel argument mismatch after the flash-attention CUDA signature gained sink/sliding parameters. The Rust launch code still used the old argument order, which led to invalid KV-cache reads and illegal-address faults during layer-0 prefill.

## Impact

- `openai/gpt-oss-20b` now loads and serves through the CUDA engine path.
- GPT-OSS is now documented in the README supported-architectures table.
- CPU/mock GPT-OSS remains intentionally limited and still reports targeted unsupported errors there.

## Validation

Automated:
- `cargo test -p rvllm-model-runner gpt_oss --lib`
- `cargo test -p rvllm-model-runner gpt_oss_moe_forward_mxfp4_smoke --lib --features cuda`
- `cargo test -p rvllm-worker --features cuda gpt_oss_disables_cuda_graphs --lib`
- `cargo check -p rvllm-model-runner --features cuda`
- `cargo check -p rvllm-worker --features cuda`
- `cargo check -p rvllm-engine --features cuda`
- `cargo build -p rvllm-server --features cuda`
- `cargo build --release -p rvllm-server --features cuda`

Manual runtime validation against local weights at `/data/models/openai/gpt-oss-20b`:
- `target/release/rvllm serve --model /data/models/openai/gpt-oss-20b --host 127.0.0.1 --port 8010 --max-model-len 2048 --gpu-memory-utilization 0.90 --log-level info`
- `GET /health` returned `ok`
- `GET /v1/models` returned `/data/models/openai/gpt-oss-20b`
- `POST /v1/chat/completions` completed successfully end-to-end on the GPT-OSS path

## Current limitation

On the validated local single-GPU 24 GB RTX 3090 setup, GPT-OSS decode does not currently use CUDA graph replay. During GPT-OSS graph-decode preparation, expert tensor upload still fails with `CUDA_ERROR_OUT_OF_MEMORY` (`model.layers.23.mlp.experts.down_proj_blocks` in the observed run), so the worker disables graph replay and falls back to eager decode. The model still serves correctly in that mode, but this PR does not yet make GPT-OSS graph-safe on that VRAM budget.

### CODE REVIEW

**Comment by gracee3** [comment]
Split out the mergeable GPT-OSS architecture/plumbing slice into a separate branch as requested.

New PR: #13
https://github.com/m0at/rvllm/pull/13

Branch: `gracee3:feature/gpt-oss-architecture-plumbing`
Commit: `3bfb3b2`

Included there:
- `gpt_oss.rs` architecture + registration
- GPT-OSS config parsing / loader / worker config plumbing
- MXFP4 expert kernel path

Deferred from this PR:
- `gpu_layer.rs` structural refactor
- `gpu_runner.rs` fusion rework
- broader `gpu_worker.rs` loading refactor

Validation on the split branch:
- `cargo check -p rvllm-model-runner --features cuda`
- `cargo check -p rvllm-worker --features cuda`
- `cargo check -p rvllm-engine --features cuda`

**Comment by gracee3** [comment]
PR #6 is now reduced to the deferred follow-up/refactor scope and force-pushed at `c583276`.

Current file set:
- `crates/rvllm-model-loader/src/gpu_loader.rs`
- `crates/rvllm-model-loader/src/gpu_weights.rs`
- `crates/rvllm-model-runner/src/gpu_layer.rs`
- `crates/rvllm-model-runner/src/gpu_runner.rs`
- `crates/rvllm-model-runner/src/runner.rs`
- `crates/rvllm-worker/src/config.rs`
- `crates/rvllm-worker/src/gpu_worker.rs`

The architecture/plumbing slice is split out separately in #13.

Validation on the reduced branch:
- `cargo check -p rvllm-model-runner --features cuda`
- `cargo check -p rvllm-worker --features cuda`
